app/packages/registration-schema/page-manager.ts

```diff
@@ -0,0 +1,55 @@
+import EmberObject from '@ember/object';
+import Changeset from 'ember-changeset';
+import lookupValidator from 'ember-changeset-validations';
+import { ChangesetDef } from 'ember-changeset/types';
+import {
+    buildValidation,
+    getSchemaBlockGroups,
+    ResponseValue,
+    SchemaBlock,
+    SchemaBlockGroup,
+} from 'ember-osf-web/packages/registration-schema';
+import { PageResponse } from 'ember-osf-web/packages/registration-schema/page-response';
+
+export class PageManager extends EmberObject {
+    changeset: ChangesetDef;
+    schemaBlockGroups: SchemaBlockGroup[];
+    pageResponses: Record<string, ResponseValue>;
+    pageHeadingText: string;
+    isVisited: boolean;
+
+    constructor(pageSchemaBlocks: SchemaBlock[], registrationResponses: PageResponse) {
+        super();
+        this.schemaBlockGroups = getSchemaBlockGroups(pageSchemaBlocks);
+        this.pageHeadingText = this.schemaBlockGroups[0].labelBlock!.displayText!;
+        this.pageResponses = {};
+        this.isVisited = false;
+        for (const group of this.schemaBlockGroups) {
+            const groupKey = group.registrationResponseKey;
+            if (groupKey && groupKey in registrationResponses) {
+                this.pageResponses[groupKey]
+                    = registrationResponses[groupKey];
+                this.isVisited = true;
```

Comment:
Since one could visit a page (say with all questions optional) and then navigate away w/o leaving any response, we can't rely on the `registrationResponses` object to to set `isVisited`. IMO, this is best taken care of by the `DraftRegistrationManager`. Soon as the user navigates to a page,  `DraftRegistrationManager` can mark it as visited. Same concern also holds if the page has some required questions.

Reply:
The plan was that if any of the `groupKeys` is present in the `registrationResponses`, we initialize `isVisited` to `true` and DraftRegistrationManager will update `isVisited` when the page is actually visited.

Reply:
@fabmiz your concern is valid, but i think this approach is ok -- the two things `DraftRegistrationManager` needs to care about for `isVisited` are:
1. when setting/saving changes to `draft.registrationResponses`, don't set anything for pages with `!isVisited` (not even `undefined` -- the page's keys should be missing)
2. when the user visits a page, set `page.isVisited = true` (or maybe that'd be cleaner calling a method `page.setVisited()`? either way, just mark the page as visited)

Reply:
The presence of `registrationResponseKey` in `registrationResponses` implies the page's been visited, however it's preferable to have one manager responsible for this. There is no use of doing it here given it's already set to `true` after transition to current page.

Reply:
@fabmiz @adlius this cross-PR coordination seems more trouble than it's worth -- how about let's merge this largely as-is, then if you need to update PageManager in the DraftRegistrationManager PR, that's cool

Reply:
(yall could even pair on it!)

Reply:
Right! These aren't blockers.